src/radical/entk/appman/wfprocessor.py

```diff
@@ -299,6 +301,8 @@ def _update_dequeued_task(self, deq_task):
 
                 # Next search across all stages of a matching
                 # pipelines
+                stage = None
```

Comment:
That should never happen, and I would suggest to replace this with `assert(pipe.stages)` if it did happen.  In that case we got an orphaned task which does not belong to any stage which we need to flag.  We could also add an `assert(stage)` after the next for loop to ensure that the stage is correctly identified.  In general (but out of scope for this PR), we should replace the search with a (dict) lookup, as that will really not scale well for large workflows...

Reply:
I am perfectly fine with changing it. I did it due to this message from pylint:
```
src/radical/entk/appman/wfprocessor.py:350:19: W0631: Using possibly undefined loop variable 'stage' (undefined-loop-variable)
```

To me it makes sense. The `if` in that line ([here](https://github.com/radical-cybertools/radical.entk/blob/fix/unittests/src/radical/entk/appman/wfprocessor.py#L350)) makes the assumption that the for loop in [line 306](https://github.com/radical-cybertools/radical.entk/blob/fix/unittests/src/radical/entk/appman/wfprocessor.py#L306) finds a stage or it will return the last stage.

> We could also add an assert(stage) after the next for loop to ensure that the stage is correctly identified.

This will not fix the warning from pylint. Also, I do not thing it will work. Even if there is a orphan task, the last stage will populate the `stage` variable since that is how the for loop ended.

I am open to suggestions.

Reply:
Yeah, please do add the asserts.  As for the warning: it's useless in this case, but adding an exception is also not useful, so let's stick with it.

Reply:
Done